enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -1793,9 +1793,44 @@ func (p *PebbleCache) Reader(ctx context.Context, r *rspb.ResourceName, uncompre
 	return pebble.ReadCloserWithFunc(rc, db.Close), nil
 }
 
+type zstdDecompressor struct {
+	d io.ReadCloser
+
+	cacheName            string
+	groupID              string
+	numDecompressedBytes int64
+}
+
+func newZstdDecompressor(cacheName string, groupID string, r io.Reader) (io.ReadCloser, error) {
```

Comment:
was surprised we needed to do all this messing with zstdDecompressor -- is it any simpler to instrument bytestream handler and inc the metrics there?